### ISSUE TITLE
Support midi files

### ISSUE THREAD

**Issue by sirkitree** [Frooxius/NeosPublic#69]
WAV files are great, but they're huge. I like to do chiptunes, and typically use them in midi format as the file size is about 80x smaller than wav. When I tried to import a midi file into my [trip] world, it wouldn't allow it, so I'm assuming there is no support for it currently. This would greatly relieve my cloud storage if it was supported. Thanks for any consideration you might give this request in the future!

**Comment by Frooxius** [comment]
Yeah I currently don't have a codec that would support them (or module files). I'm looking into switching to a native audio decoder library though, but not sure when I'll get to it.

You can use OGG Vorbis files though, they offer quite good compression with good quality. Or FLAC.

**Comment by Enverex** [comment]
We now have VLC support and VLC natively supports MIDI files (and module files which were mentioned, now already work). AFAIK VLC needs to be pointed to a soundfont but that's it - as far as the standalone version goes, not sure how much work it is with the Unity version.

**Comment by Frooxius** [comment]
We haven't upgraded to the official libVLC integration yet. I'm not fully sure whether it's possible to pass the soundfont to that version, but from last time I looked with the 3rd party library it wasn't really feasible. Since it's going away anyways, I don't plan on investigating it more until the swap occurs.

**Comment by dfgHiatus** [comment]
Hey all, I have a workaround for this: https://github.com/dfgHiatus/MIDIImporter
